src/components/trade/PriceSuggestions/PriceSuggestionItem.tsx

```diff
@@ -28,18 +29,39 @@ interface FormattedPrices {
 
 const LOW_PRICE_FLOOR = new BigNumber('0.0001')
 
-export function formatPriceToPrecision(price: BigNumber, useThreshold = false): string {
-  return price.gt(LOW_PRICE_FLOOR) || !useThreshold
-    ? price.toFixed(PRICE_ESTIMATION_PRECISION)
-    : '< ' + LOW_PRICE_FLOOR.toString()
+function formatPriceWithFloor(price: BigNumber): string {
+  const priceAsBN = parseAmount(price.toString(10), 18)
+  if (!priceAsBN || priceAsBN.isZero()) return 'N/A'
+
+  const displayPrice = formatAmount({ amount: priceAsBN, precision: 18, decimals: DEFAULT_DECIMAL_PLACES })
+  return price.gt(LOW_PRICE_FLOOR) ? displayPrice : '< ' + LOW_PRICE_FLOOR.toString(10)
+}
+
+interface GetPriceFormatted {
+  price: BigNumber | null
+  isPriceInverted: boolean
 }
 
-function getPriceFormatted(price: BigNumber | null, isPriceInverted: boolean): FormattedPrices {
+function getPriceFormatted({ price, isPriceInverted }: GetPriceFormatted): FormattedPrices {
   if (price) {
-    const inversePriceLabel = formatPriceToPrecision(invertPrice(price), true)
-    const priceLabel = formatPriceToPrecision(price, true)
-    const inversePriceValue = invertPrice(price).toString(10)
-    const priceValue = price.toString(10)
+    const inversePriceLabel = formatPriceWithFloor(invertPrice(price))
+    const priceLabel = formatPriceWithFloor(price)
+    // Use full precision for accuracy and tightest price
+    const inversePriceBN = parseAmount(invertPrice(price).toString(10), 18) as BN
+    const priceValueBN = parseAmount(price.toString(10), 18) as BN
+
+    // Have to explicitly pass full precision when using object params
+    const inversePriceValue = formatAmountFull({
+      amount: inversePriceBN,
+      precision: 18,
+      thousandSeparator: false,
+    })
+    // Have to explicitly pass full precision when using object params
+    const priceValue = formatAmountFull({
+      amount: priceValueBN,
+      precision: 18,
```

Comment:
🔢✨🧙‍♂️

Reply:
mm maybe the comment isnt clear enough but formatAmountFull by `default` should pass in `18` precision but when using object params it requires it, it's a bug

Reply:
fixed, using token decimals